integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py

```diff
@@ -50,3 +51,52 @@ def embed(self, data: List[str], progress_bar=True, **kwargs) -> List[List[float
         ):
             embeddings.append(np_array.tolist())
         return embeddings
+
+
+class _FastembedSparseEmbeddingBackendFactory:
+    """
+    Factory class to create instances of fastembed sparse embedding backends.
+    """
+
+    _instances: ClassVar[Dict[str, "_FastembedSparseEmbeddingBackend"]] = {}
+
+    @staticmethod
+    def get_embedding_backend(
+        model_name: str,
+        cache_dir: Optional[str] = None,
+        threads: Optional[int] = None,
+    ):
+        embedding_backend_id = f"{model_name}{cache_dir}{threads}"
+
+        if embedding_backend_id in _FastembedSparseEmbeddingBackendFactory._instances:
+            return _FastembedSparseEmbeddingBackendFactory._instances[embedding_backend_id]
+
+        embedding_backend = _FastembedSparseEmbeddingBackend(
+            model_name=model_name, cache_dir=cache_dir, threads=threads
+        )
+        _FastembedSparseEmbeddingBackendFactory._instances[embedding_backend_id] = embedding_backend
+        return embedding_backend
+
+
+class _FastembedSparseEmbeddingBackend:
+    """
+    Class to manage fastembed sparse embeddings.
+    """
+
+    def __init__(
+        self,
+        model_name: str,
+        cache_dir: Optional[str] = None,
+        threads: Optional[int] = None,
+    ):
+        self.model = SparseTextEmbedding(model_name=model_name, cache_dir=cache_dir, threads=threads)
+
+    def embed(self, data: List[List[str]], **kwargs) -> List[Dict[str, Union[List[int], List[float]]]]:
+        # The embed method returns a Iterable[SparseEmbedding], so we convert it to a list of dictionaries.
+        # Each dict contains an `indices` key containing a list of int and an `values` key containing a list of floats.
+        sparse_embeddings = [sparse_embedding.as_object() for sparse_embedding in self.model.embed(data, **kwargs)]
+        for embedding in sparse_embeddings:
+            embedding["indices"] = embedding["indices"].tolist()
+            embedding["values"] = embedding["values"].tolist()
```

Comment:
I would like to introduce a progress bar as done with the dense Embedders, but I'm not sure I did it right (results do not seem good).
@lambda-science can you have a look?
@Anush008 if you have ideas about this, please help.

Reply:
Agree with the progress bar ! I can test some things maybe

Reply:
Committed a solution exactly like the dense embedding implementation of Fastembed as
```python
    def embed(
        self, data: List[List[str]], progress_bar=True, **kwargs
    ) -> List[Dict[str, Union[List[int], List[float]]]]:
        # The embed method returns a Iterable[SparseEmbedding], so we convert it to a list of dictionaries.
        # Each dict contains an `indices` key containing a list of int and an `values` key containing a list of floats.

        sparse_embeddings = []
        sparse_embeddings_iterable = self.model.embed(data, **kwargs)
        for sparse_embedding in tqdm(
            sparse_embeddings_iterable, disable=not progress_bar, desc="Calculating sparse embeddings", total=len(data)
        ):
            sparse_embedding_obj = sparse_embedding.as_object()
            sparse_embedding_obj["indices"] = sparse_embedding_obj["indices"].tolist()
            sparse_embedding_obj["values"] = sparse_embedding_obj["values"].tolist()
            sparse_embeddings.append(sparse_embedding_obj)

        return sparse_embeddings
```